examples/plugins/afk.js

```diff
@@ -0,0 +1,50 @@
+// Plugins are loaded using the bot.loadPlugin() function
+// The load plugin function excepts one function as its only parameter
+// The plugin runs the function passing it two parameter
+// The former is the bot object and the latter is the bot option
+// the options contain version player info and other plugins
+// Here is an example AFK plugin
+
+// Each plugin should have a inject function
+// This is the function that will be put in bot.loadPlugin
+function inject (bot, option) {
+// It is a good idea to make a namespace for your plugin
+  bot.afk = {}
+
+  let afkInterval, rotation
+  // This namespace can be used to store some variuabels for user reffrence.
+  bot.afk.status = 'Idel'
+
+  // All your plugin functions should be in the name space other than the logic.
+  bot.afk.start = async function () {
+    afkInterval = setInterval(async () => {
```

Comment:
whats the point in this being async, awaiting before setting the boolean to true is pointless

Reply:
This way is more precise, so I don't want to change it. Think about it if the server has a lot of lag, the bloolen would be assigned too early, and if the lag continues it will cause issues with too many requests going on.

does that make any sence?

Reply:
no, not really, a server having 3 seconds of lag between look packets is unheardof

Reply:
I had a 4 minute delay between pearl meeting the ground and me getting teleported there today, guess what was happening with packets. The fact that this won't stop the bot from sending look packets is a different question:
Open browser console and try this
`setInterval(async () => {setTimeout(console.log, 2500, 'this will execute every 500ms with initial delay of 3s')}, 500)`